File: Anul 2/MAP/Sem9/seminar9/seminar9/domain/TaskContainerFactory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace seminar9.domain
{
    class TaskContainerFactory : IFactory
    {
        public IContainer CreateContainer(Strategy startegy)
        {
            if (startegy == Strategy.LIFO) return new StackContainer();
            else return null; //return new QueueContainer();
        }
    }
}

File: Anul 2/MAP/Seminar 1/src/com/company/sem1/factory/TaskContainerFactory.java
package com.company.sem1.factory;

import com.company.sem1.container.Container;
import com.company.sem1.container.QueueContainer;
import com.company.sem1.container.StackContainer;
import com.company.sem1.container.Strategy;

public class TaskContainerFactory implements Factory{
    private static TaskContainerFactory instance= null;
    //public static final TaskContainerFactory INSTANCE = new TaskContainerFactory();

    @Override
    public Container createContainer(Strategy strategy) {
        if(strategy == Strategy.LIFO)
            return new StackContainer();
        else
            return new QueueContainer();
    }
    private TaskContainerFactory(){

    }
    public static TaskContainerFactory getInstance(){
        if(instance==null)
            instance= new TaskContainerFactory();
        return instance;
    }
}
